src/lib/Repository/DecoratedLocationService.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use EzSystems\HybridPlatformUi\Decorator\LocationDecorator;
+
+/**
+ * Service for loading locations with additional data not provided by the original API.
+ * Returns decorated location objects to facilitate storing the additional data without modifying the original object.
+ */
+class DecoratedLocationService
+{
+    /**
+     * @var LocationService
+     */
+    private $locationService;
+
+    /**
+     * @var PathService
+     */
+    private $pathService;
+
+    public function __construct(LocationService $locationService, PathService $pathService)
+    {
+        $this->locationService = $locationService;
+        $this->pathService = $pathService;
+    }
+
+    /**
+     * Load decorated locations.
```

Comment:
I'd highlight a bit more that it loads _all_ the locations of a given ContentInfo, and decorates them with extra info. I'd _maybe_ even add which info.

There is an overhead when using this service. I'd really prefer if no developers use it without a good reason :)